ngraph/core/src/partial_shape.cpp

```diff
@@ -34,15 +34,15 @@ PartialShape::PartialShape(const Shape& shape)
 {
 }
 
-PartialShape::PartialShape(bool rank_is_static, const std::vector<Dimension>& dimensions)
+PartialShape::PartialShape(bool rank_is_static, std::vector<Dimension> dimensions)
     : m_rank_is_static(rank_is_static)
-    , m_dimensions(dimensions)
+    , m_dimensions(std::move(dimensions))
```

Comment:
Previously we did a copy in the constructor, now we do a copy before the `dimensions` are passed to the constructor and then perform the move construction. Isn't this less efficient or am I missing something?

Reply:
If we pass r-value to the constructor we don't create copy before constructor is called. If we want to have supper efficient implementation we need two constructors:
```
PartialShape(const std::vector<Dimension>& dimensions);
PartialShape(std::vector<Dimension>&& dimensions);
```
but in this situation we need to maintain two implementation.

Reply:
e.g.:
https://github.com/openvinotoolkit/openvino/pull/6779/files#diff-db882ebabc1f33eb7ee439df43f84f81c7170a5e56b9b9c948d9ad28bfdf9b6bR390